VersionControlUI/src/nodes/DiffComparisonPair.cpp

```diff
@@ -25,14 +25,99 @@
  **********************************************************************************************************************/
 #include "DiffComparisonPair.h"
 
+#include "ModelBase/src/nodes/composite/CompositeNode.h"
+
 namespace VersionControlUI
 {
 
 DEFINE_NODE_TYPE_REGISTRATION_METHODS(DiffComparisonPair)
 
+DiffComparisonPair::DiffComparisonPair(Model::Node* oldVersionNode, Model::Node* newVersionNode) : Super{}
+{
+	oldVersionNode_ = oldVersionNode;
+	newVersionNode_ = newVersionNode;
+
+	setUpObjectPath();
+
+	setUpComponentType();
+
+}
+
+void DiffComparisonPair::setUpObjectPath()
+{
+	QString oldVersionObjectPath = computeObjectPath(oldVersionNode_);
+	QString newVersionObjectPath = computeObjectPath(newVersionNode_);
+	if (oldVersionObjectPath == newVersionObjectPath)
+	{
+		singleObjectPath_ = new Model::Text{oldVersionObjectPath};
+		twoObjectPathsDefined_ = false;
+	}
+	else if (oldVersionObjectPath == "" || newVersionObjectPath == "")
+	{
+		auto objectPath = oldVersionObjectPath == "" ? newVersionObjectPath : oldVersionObjectPath;
+		singleObjectPath_ = new Model::Text{objectPath};
+		twoObjectPathsDefined_ = false;
+	}
+	else
+	{
+		oldVersionObjectPath_ = new Model::Text{oldVersionObjectPath};
+		newVersionObjectPath_ = new Model::Text{newVersionObjectPath};
+		twoObjectPathsDefined_ = true;
+	}
+}
+
+QString DiffComparisonPair::computeObjectPath(Model::Node* node)
+{
+	if (!node) return "";
+	auto parent = node->parent();
+	QString objectPath = "";
+	while (parent)
+	{
+		if (parent->definesSymbol())
+			objectPath.prepend(parent->symbolName() + "/");
+		parent = parent->parent();
+	}
+
+	return objectPath;
+}
+
+void DiffComparisonPair::setUpComponentType()
+{
+	Model::Node* nodeToComputeComponentType = nullptr;
+	QString componentType = "unknown";
+
+	if (oldVersionNode_)
+		nodeToComputeComponentType = oldVersionNode_;
+	else
+		nodeToComputeComponentType = newVersionNode_;
+
+	auto searchCompositeNode = nodeToComputeComponentType;
+
+	while (searchCompositeNode && !DCast<Model::CompositeNode>(searchCompositeNode))
+		searchCompositeNode = searchCompositeNode->parent();
+
+	if (searchCompositeNode)
+	{
+		Model::CompositeNode* compositeNode = DCast<Model::CompositeNode>(searchCompositeNode);
+		if (compositeNode == nodeToComputeComponentType)
+		{
+			componentType = compositeNode->typeName();
```

Comment:
Oh I didn't expect to see this here. Is it really useful to show this? Ideally the object we show for the diff is some bigger object, where the type is clear from the visualization.

Do you have a use case in mind where, it's useful to show the type name?